docs/chapters/architecture/SOTA-client.rst

```diff
@@ -64,3 +64,115 @@ Glossary
+requirements, has a fully-featured deployment panel and a handy Yocto layer.
+Mender is easy to integrate to an embedded Linux system but at the cost of its
+lack of flexibility. Mender imposes an A/B scheme with two additional partitions
+for bootloader and data. The kernels also have to be located in the A and B
+partitions as files. The goal of PELUX being to serve as a baseline for various
+projects, we will prefer a more flexible solution such as one of those detailed
+below.
+
+**OSTree**: OSTree is an elegant file-based update mechanism that uses hard links
+to achieve in-place(no A/B partitioning) atomic updates. It is often described
+as a "git for operating systems". It currently benefits from a very large and
+active community. It has support for rollback  It integrates with Yocto easily.
+It was chosen by AGL for all of those reasons. However, OSTree suffers from some
+limitations if the rootfs to be upgraded is corrupted and since OSTree is only
+able to update file systems, it can not always upgrade kernels and can not flash
+other types of firmwares such as Bootloaders or MCUs. Hence, this solution may
+not be enough on its own depending on the needs of the project.
+
+**QtOTA**: This solution contains a set of scripts and QML APIs to easily
+integrate OSTree in a Yocto and Qt/QML system. QtOTA seems preferable over
+OSTree alone if the final system is tightly linked to a Qt architecture.
+
+**GENIVI SOTA (Aktualizr)**: GENIVI defined a modular architecture for Software
+Over The Air update deployment split into a SOTA Server, SOTA Client and
+installer. The SOTA server offers various deployment scenarios based a on VIN
+(vehicle identifiers) registry. The client side, whose current reference
+implementation is Aktualizr, can download any kind of data from the server and
+relay that to an installer. It is also worth noting that this implementation
+supports complex security mechanisms using Uptane and RVI. This architecture
+isn't enough on its own but can be integrated with something else to provide a
+fully featured update solution.
+
+**SWUpdate**: This tool is extremely flexible, it is even described by its
+developers as an update framework. It is fault resilient, supports atomic
+updates, fallback(with both A/B and Normal+recovery). it makes few assumptions
+regarding the base system, flashes entire compressed images, it can interface
+with complex fleet management systems such as Hawkbit, it guarantees integrity
+and authentication, offers APIs for GUI integration, is easily integrated to
+Yocto and can be extended with handlers to upgrade FPGAs, MCUs or other
+components and is well documented... SWUpdate meets the requirements of PELUX.
+
+**RAUC**: This solution is failsafe, atomic, can revert to a previous state, is
+flexible enough when it comes to partition layout, uses bundle of images that
+can be downloaded from the network or from local media, interfaces with Hawkbit,
```

Comment:
Perhaps a link to Hawkbit, or a description of what it is, would be good

Reply:
Might be worth looking at https://advancedtelematic.github.io/rvi_sota_server/ since it is purpose built for automotive and modular.

Reply:
rvi_sota_server/client haven't been updated for a long while now https://github.com/genivi/rvi_sota_client and it looks like RVI is abandoned.
You can check the "GENIVI SOTA (Aktualizr)" paragraph for some more details.
Also, we finished the Architecture Meeting with the conclusion that we wouldn't rely on GENIVI SOTA.

Reply:
Hey @jeremiah , thanks for brining this up. I believe that the github link to that work would be this one: https://advancedtelematic.github.io/rvi_sota_server/. Like @FlorentRevest mentioned, the efforts were abandoned. We can always bring back the topic and integrate our solution against the RVI if we find a future scenario for it.

Reply:
The RVI efforts were abandoned by ATS, but RVI is being used in production by JLR. ATS is moving to use TLS directly, which I think is smart. In any case, I just wanted to put GENIVI's SOTA on the radar but since you've already investigated, I think we can skip it to keep the list shorter.